plugins/catalog-import/src/api/CatalogImportClient.ts

```diff
@@ -175,7 +176,15 @@ the component will become available.\n\nFor more information, read an \
     body: string;
   }): Promise<{ link: string; location: string }> {
     const { repositoryUrl, fileContent, title, body } = options;
-
+    const parseData = YAML.parse(fileContent);
+    const {
+      metadata: { name },
+    } = parseData;
```

Comment:
It looks like we have access to the catalog API in this class, can we use the [`validateEntity` method](https://github.com/backstage/backstage/blob/patch/v1.22.0/packages/catalog-client/src/types/api.ts#L632-L642) instead? That will run it against the current catalog instance and ensure it can be imported later. Users may have customized their naming format or other formats and we'll want to check those as well IMO.

Reply:
did not know about this 
this would be good I'm not sure?
```
async submitPullRequest(options: {
    repositoryUrl: string;
    fileContent: string;
    title: string;
    body: string;
  }): Promise<{ link: string; location: string }> {
    const { repositoryUrl, fileContent, title, body } = options;
    const parseData = YAML.parse(fileContent);
    try {
      const validationResponse = await this.catalogApi.validateEntity(
        parseData,
        `url:${repositoryUrl}`,
      );
      if (!validationResponse.valid) {
        throw new Error(validationResponse.errors[0].message);
      }
      const ghConfig = getGithubIntegrationConfig(
        this.scmIntegrationsApi,
        repositoryUrl,
      );

      if (ghConfig) {
        return await this.submitGitHubPrToRepo({
          ...ghConfig,
          repositoryUrl,
          fileContent,
          title,
          body,
        });
      }
      throw new Error('unimplemented!');
    } catch (err) {
      throw err;
    }
  }
```
having issue with tests.

Reply:
I believe that looks good, I think for this we can just use `url:http://localhost` or similar non-value for the location. It would need to point at an existing repository and only for `Location` kinds which _shouldn't_ be used here @backstage/scaffolder-maintainers to confirm though?